### ISSUE TITLE
Microsoft C++ exception: CGAL::Uncertain_conversion_exception at memory location 0x000000029FEFA6D0 . and program cannot end

### ISSUE THREAD

**Issue by wangbowenll** [CGAL/cgal#3852]
_Please use the following template to help us solving your issue._

## Issue Details
Dear researcher
       I want to mesh the subdomain defined by some surface patch. the input domain is similar to the example domain  which is named File [Mesh_3/mesh_polyhedral_complex.cpp.](https://doc.cgal.org/latest/Mesh_3/index.html#Mesh_3PolyhedralComplex)
i input the .off file created by myself.it is composed of 6 surface patchs. and the example also has six surface patches. i think  i can get the mesh,because my domain is very similar to the example,while i failed. when running the code,exception is occured,it is CGAL::Uncertain_conversion_exception. i try solve it with many methods,finally i failed.i google it,someone said it is a interior exception so that you can ignore it.  but when running i choose the option of  “cannot Interrupt when this exception type is thrown” in vs 2017   so that the program can continue  running. but atter running 30 hours,the program is not end.so i think there must be error in my program,but i really cannot know what it is.
please  help me. i had spent 10 days in doing it.I would appreciate any input on this matter. 

_Describe your issue. Please be specific (compilation error, runtime error, unexpected behavior, wrong results, etc.)._

## Source Code

the  CGAL example is [https://doc.cgal.org/latest/Mesh_3/Mesh_3_2mesh_polyhedral_complex_8cpp-example.html](url)

my code is
```
#define CGAL_CONCURRENT_MESH_3
#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>

#include <CGAL/Mesh_triangulation_3.h>
#include <CGAL/Mesh_complex_3_in_triangulation_3.h>
#include <CGAL/Mesh_criteria_3.h>

#include <CGAL/Polyhedral_complex_mesh_domain_3.h>
#include <CGAL/make_mesh_3.h>

#include <cstdlib>

// Domain
typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
typedef CGAL::Mesh_polyhedron_3<K>::type Polyhedron;
typedef CGAL::Polyhedral_complex_mesh_domain_3<K> Mesh_domain;


#ifdef CGAL_CONCURRENT_MESH_3
typedef CGAL::Parallel_tag Concurrency_tag;
#else
typedef CGAL::Sequential_tag Concurrency_tag;
#endif

// Triangulation
typedef CGAL::Mesh_triangulation_3<Mesh_domain,CGAL::Default,Concurrency_tag>::type Tr;

typedef CGAL::Mesh_complex_3_in_triangulation_3<
  Tr,Mesh_domain::Corner_index,Mesh_domain::Curve_index> C3t3;

// Criteria
typedef CGAL::Mesh_criteria_3<Tr> Mesh_criteria;

// To avoid verbose function and named parameters call
using namespace CGAL::parameters;

const char* const filenames[] = {
  "data/0123/01_CGAL.off",
  "data/0123/13_CGAL.off",
  "data/0123/20_CGAL.off",
  "data/0123/21_CGAL.off",
  "data/0123/23_CGAL.off",
  "data/0123/30_CGAL.off",
    
  
};

const std::pair<int, int> incident_subdomains[] = {
  std::make_pair(0, 1),
  std::make_pair(1, 3),
  std::make_pair(2, 0),
  std::make_pair(2, 1),
  std::make_pair(2, 3),
  std::make_pair(3, 0),
   
};

int main()
{
  const std::size_t nb_patches = sizeof(filenames) / sizeof(const char*);
  CGAL_assertion(sizeof(incident_subdomains) ==
                 nb_patches * sizeof(std::pair<int, int>));
  std::vector<Polyhedron> patches(nb_patches);
  for(std::size_t i = 0; i < nb_patches; ++i) {
    std::ifstream input(filenames[i]);
    if(!(input >> patches[i])) {
      std::cerr << "Error reading " << filenames[i] << " as a polyhedron!\n";
      return EXIT_FAILURE;
    }
  }
  // Create domain
  Mesh_domain domain(patches.begin(), patches.end(),
                     incident_subdomains, incident_subdomains+nb_patches);

  domain.detect_features(); //includes detection of borders

  // Mesh criteria
  Mesh_criteria criteria(edge_size = 20,
	  facet_angle = 25, facet_size = 30, facet_distance = 8,
	  cell_radius_edge_ratio = 3, cell_size =33);

  // Mesh generation
  C3t3 c3t3 = CGAL::make_mesh_3<C3t3>(domain, criteria,
	  no_exude(),
	  perturb(sliver_bound = 10, time_limit = 15));

  // Output
  std::ofstream medit_file("midas_3.mesh");
  c3t3.output_to_medit(medit_file);

  return EXIT_SUCCESS;
}
```

![exception](https://user-images.githubusercontent.com/43980713/56088081-ea031580-5eab-11e9-96b9-a08dc308937a.png)

the .off file 
[https://github.com/wangbowenll/.off-file](url)    the file named 0123 is .rar file.

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits):windows 64 bits
* Compiler: vs 2017 community and vc14.1 (in cmake     visual studio 15 2017 win 64)
* Release or debug mode: debug mode
* Specific flags used (if any):  
* CGAL version: CGAL 4.13
* Boost version:1.67
* Other libraries versions if used (Eigen, TBB, etc.):TBB +Engin

**Comment by wangbowenll** [comment]
@lrineau dear researcher
What am I missing? It must be something really basic. CGAL is a very reliable library that works well on complex data; I think the Mesh criteria also is right.

**Comment by janetournois** [comment]
Dear @wangbowenll 
I tried to download your input .off file but the link seems to be broken.
Can you please have a look at the PR #3873 and check if it solves this issue for you?

**Comment by wangbowenll** [comment]
> Dear @wangbowenll
> I tried to download your input .off file but the link seems to be broken.
> Can you please have a look at the PR #3873 and check if it solves this issue for you?

Dear @janetournois 
Thanks for your guide. I have change some code at the header file located in my computer after looking the  PR #3873. while after running 10 hours.it still failed. The program seemed to terminal correctly,but
no any .mesh file is generated. So i am so confused. I check my model. I think it is ok. I have re-updown the .off file.  It is a package.

[0123 (2).zip](https://github.com/CGAL/cgal/files/3115054/0123.2.zip)

**Comment by janetournois** [comment]
Hi @wangbowenll 

In the above example, I think the incident subdomains should be listed this way, wrt to orientation of surfaces :
```
const std::pair<int, int> incident_subdomains[] = {
  std::make_pair(0, 1),
  std::make_pair(3, 1),
  std::make_pair(0, 2),
  std::make_pair(1, 2),
  std::make_pair(3, 2),
  std::make_pair(0, 3),
};
```

Using this set of pairs, I can still reproduce your issue and will try to fix it.

**Comment by janetournois** [comment]
Hi @wangbowenll 

I just checked your surfaces, and it appears that they are not exactly conformal.
Take surfaces `23_CGAL.off` and `30_CGAL.off`. I tried to stitch them together using [`copy_face_graph()`](https://doc.cgal.org/latest/BGL/group__PkgBGLHelperFct.html#ga2d3eb044bfe42eab566779ce32216c11) and [`stitch_borders()`](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__repairing__grp.html#gabdf5abc4d0f51055bf12afb00a128abc) on the resulting polyhedral surface, and it appears that borders cannot be stitched together - most probably because vertices coordinates are not all exactly equal.

It is the reason why the feature detection process detects corners (shown as black points in the snapshot below) where there should not be any, and the reason why meshing seems not to terminate.
![image](https://user-images.githubusercontent.com/11311305/56895835-70158380-6a8a-11e9-90c6-c6f267ac0003.png)

You can try to use the upcoming snapping function, available in PR #3266 to fix your input before restarting the mesh process.
